File: src/ARCtrl/WebRequest/WebRequest.Node.fs
module ARCtrl.WebRequestHelpers.NodeJs

open Fable.Core
open Fable.SimpleHttp

open Fable.Core.JsInterop

open Fable.SimpleHttp

open Fable.Core
open Fable.Core.JsInterop
open Fetch

let isNode() =
    emitJsExpr
        ()
        """typeof process !== "undefined" &&
process.versions != null &&
process.versions.node != null;"""

//// From here: https://github.com/fable-compiler/fable3-samples/blob/25ea2404b28c897988b144f0141bc116da292679/nodejs/src/App.fs#L7

let downloadFile url =
    fetch url []
    |> Promise.bind (fun res -> res.text()) // get the result
    |> Promise.map (fun txt -> // bind the result to make further operation
        txt
    )
    |> Async.AwaitPromise

File: js/WebRequest/WebRequest.Node.js
import { awaitPromise } from "../fable_modules/fable-library-js.4.22.0/Async.js";
import { fetch$ } from "../fable_modules/Fable.Fetch.2.6.0/Fetch.fs.js";
import { empty } from "../fable_modules/fable-library-js.4.22.0/List.js";

export function isNode() {
    return typeof process !== "undefined" &&
process.versions != null &&
process.versions.node != null;;
}

export function downloadFile(url) {
    let pr_1, pr;
    return awaitPromise((pr_1 = ((pr = fetch$(url, empty()), pr.then((res) => res.text()))), pr_1.then((txt) => txt)));
}

